src/hooks/custom-hooks/useIsP2PBlocked.ts

```diff
@@ -0,0 +1,52 @@
+import { useMemo } from 'react';
+import { getBlockedType } from '@/utils';
+import { useGetAccountStatus } from '@deriv-com/api-hooks';
+import { api } from '..';
+
+// TODO: add other blocked statuses when implementing high risk scenarios
+/**
+ * A custom hook that returns the P2P blocked status and the status type.
+ * @returns {object} { isP2PBlocked: boolean, status: string }
+ */
+const useIsP2PBlocked = () => {
+    const { data: accountStatus } = useGetAccountStatus();
+    const { data: activeAccountData } = api.account.useActiveAccount();
+
+    const isAccountBlocked = useMemo(() => {
+        if (!accountStatus) return false;
+
+        const isPermBan = accountStatus.p2p_status === 'perm_ban';
+        const isSystemMaintenance = accountStatus?.cashier_validation?.includes('system_maintenance');
+        const isCashierLocked = accountStatus.status.includes('cashier_locked');
+
+        return isPermBan || isSystemMaintenance || isCashierLocked;
+    }, [accountStatus]);
+
+    const accountBlockStatus = useMemo(() => {
+        if (!accountStatus) return '';
+
+        if (accountStatus?.cashier_validation?.includes('system_maintenance')) return 'systemMaintenance';
+        if (accountStatus.status.includes('cashier_locked')) return 'cashierLocked';
+        if (accountStatus.p2p_status === 'perm_ban') return 'p2pBlocked';
+
+        return '';
+    }, [accountStatus]);
+
+    const isP2PNotSupported = useMemo(() => {
+        if (!activeAccountData) return false;
+
+        return activeAccountData.is_virtual || activeAccountData.currency !== 'USD';
```

Comment:
isn't this check already handled in getBlockedType?

Reply:
for this, we are using this to determine whether or not we show BlockedScenarios component using isP2PBlocked flag. though perhaps we could just check if type was returned from blockedType.

Reply:
yeah.. is_virtual will have demo and the other will have nonUSD. so u can just check for the type.. so i dont think this function is needed.. can check for the blockedType value where this function is used